src/saltext/github/modules/github.py

```diff
@@ -1886,4 +1897,318 @@ def _query(
+    if rule_id:
+        kwargs["ruleset_id"] = kwargs.get("ruleset_id", None)
+    if page:
+        kwargs["per_page"] = kwargs.get("per_page", None)
+
+    return kwargs
+
+
+def _check_params(profile, **kwargs):
+    """
+    Helper function to check for params values on command line
+    and in config file
+
+    profile
+        The name of the profile configuration to use.
+    """
+
+    if not kwargs.get("ruleset_type"):
+        kwargs["ruleset_type"] = _get_config_value(profile, "ruleset_type")
+    if kwargs["ruleset_type"] not in ["repo", "org"]:
+        raise CommandExecutionError
+
+    if kwargs["ruleset_type"] == "repo":
+        kwargs.pop("org_name")
+
+    if kwargs["ruleset_type"] == "org":
+        kwargs.pop("repo_name")
+        kwargs.pop("owner")
+
+    for param in kwargs:  # pylint: disable=consider-using-dict-items
+        if not kwargs[param]:
+            if param == "header_dict":
+                try:
+                    kwargs[param] = _get_config_value(profile, param)
+                except CommandExecutionError:
+                    kwargs[param] = {}
+            elif param == "per_page":
+                try:
+                    kwargs[param] = _get_config_value(profile, param)
+                except CommandExecutionError:
+                    kwargs[param] = None
+            else:
+                kwargs[param] = _get_config_value(profile, param)
+    return kwargs
+
+
+def _format_action(rule_id=False, **kwargs):
+    """
+    Helper function to format action
+    """
+    if kwargs["ruleset_type"] == "repo":
+        action = "/".join(["repos", kwargs["owner"], kwargs["repo_name"], "rulesets"])
+    else:
+        action = "/".join(["orgs", kwargs["org_name"], "rulesets"])
+    if rule_id:
+        action = action + "/" + str(kwargs["ruleset_id"])
+    return action
+
+
+def get_ruleset(profile="github", **kwargs):
+    """
+    Get information of specific ruleset.
+
+    profile
+        The name of the profile configuration to use. Defaults to ``github``.
+    ruleset_type
+        The type of ruleset. 'repo' or 'org'.
+    ruleset_id
+        Id of the ruleset.
+    owner
+        The account owner of the repository.
+    repo_name
+        The name of the repo.
+    org_name
+        The name of the organization.
+    header_dict
+        Headers to pass to Github API.
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt myminion github.get_ruleset
+        salt myminion github.get_ruleset ruleset_id=1
+    """
+
+    kwargs = _param_dict(rule_id=True, **kwargs)
+    params = _check_params(profile, **kwargs)
+    action = _format_action(rule_id=True, **params)
+
+    try:
+        ret = _query(profile, action, header_dict=params["header_dict"])
+        if not ret.get("error"):
+            return ret["dict"]
+        else:
+            ret["comment"] = f"GitHub Response Status Code: {ret.get('error')}"
+            ret["result"] = False
```

Comment:
**issue (blocking):** this will probably be caught by the linter, but it's an unnecessary `else`. any time you're returning early, you won't reach the code below it anyway

```suggestion
        if not ret.get("error"):
            return ret["dict"]

        ret["comment"] = f"GitHub Response Status Code: {ret.get('error')}"
        ret["result"] = False
```